src/types.rs

```diff
@@ -75,15 +77,41 @@ impl TxLogResponse {
         let edn = parse_edn(&resp).unwrap();
         Self {
             tx___tx_id: edn[":crux.tx/tx-id"].to_string().parse::<usize>().unwrap(), 
-            tx___tx_time: edn[":crux.tx/tx-time"].to_string()
+            tx___tx_time: edn[":crux.tx/tx-time"].to_string(),
+            tx__event___tx_events: edn.get(":crux.tx.event/tx-events").map_or(None, |e| Some(e.to_string())),
         }
     }
 
     #[cfg(test)]
     pub fn default() -> Self {
         Self {
             tx___tx_id: 8usize, 
-            tx___tx_time: "2020-07-16T21:53:14.628-00:00".to_string()
+            tx___tx_time: "2020-07-16T21:53:14.628-00:00".to_string(),
+            tx__event___tx_events: None
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+#[allow(non_snake_case)]
+/// Definition for the response of the `GET` on `tx-log` endpoint
+pub struct TxLogsResponse {
+    tx_events: Vec<TxLogResponse> 
+}
+
+impl TxLogsResponse {
+    pub fn deserialize(resp: String) -> Self {
+        let edn = parse_edn(&resp).unwrap();
+        Self {
+            tx_events: edn.iter().unwrap()
+                .map(|e| 
+                    TxLogResponse {
```

Comment:
Isn't it better to implement `From<Edn>` to `TxLogResponse`? This logic seems like it and it could be replaced to `.map(|e| TxLogResponse::from(e))`

Reply:
0db3d94